src/structures/Message.js

```diff
@@ -418,6 +422,18 @@ class Message extends Base {
     );
   }
 
+  /**
+   * The Message this crosspost/reply/pin add references, if cached
+   * @type {?Message}
+   * @readonly
+   */
+  get referencedMessage() {
+    if (!this.reference) return null;
+    const referenceChannel = this.client.channels.resolve(this.reference.channelID);
+    if (!referenceChannel) return null;
+    return referenceChannel.messages.resolve(this.reference.messageID);
```

Comment:
would it be better to just use `.cache.get` rather than resolve here? since they're the ids it seems pretty useless to use resolve

Reply:
Resolve does the same thing anyway, and seemed more consistent with other internal usage